src/garage/_dtypes.py

```diff
@@ -167,3 +167,199 @@ def __new__(cls, env_spec, observations, actions, rewards, terminals,
         return super().__new__(TrajectoryBatch, env_spec, observations,
                                actions, rewards, terminals, env_infos,
                                agent_infos, lengths)
+
+class SamplesBatch(
```

Comment:
i think you should focus on defining a good `Sample` dtype and then consider later if defining a collection dtype is even necessary.